src/WrappedHttpHandler.php

```diff
@@ -95,9 +95,21 @@ private function parseResponse(
     ) {
         $parser = $this->parser;
         $status = $response->getStatusCode();
-        $result = $status < 300
-            ? $parser($command, $response)
-            : new Result();
+        try {
+            $result = $status < 300
+                ? $parser($command, $response)
+                : new Result();
+        } catch (\Exception $e) {
+            return new Promise\RejectedPromise($this->parseDeserializationError(
```

Comment:
Why is this code necessary? It also violates the return type as we need to return a ResultInterface here not a promise.